homeassistant/components/withings/config_flow.py

```diff
@@ -81,6 +81,12 @@ async def async_step_profile(self, data: dict[str, Any]) -> FlowResult:
         )
 
     async def async_step_reauth(self, data: Mapping[str, Any]) -> FlowResult:
+        """Prompt user to re-authenticate."""
+        return await self.async_step_reauth_confirm()
```

Comment:
`data` can never be `None`, so it would always jump straight to `async_step_user`
I have adjusted this to add the confirmation step.